projects/mercury/src/metadata/metadata-sources/InternalMetadataSourceContext.js

```diff
@@ -0,0 +1,35 @@
+import React, {useContext} from 'react';
+import MetadataSourceContext from './MetadataSourceContext';
+import {DEFAULT_METADATA_VIEW_MENU_LABEL} from '../../constants';
+import {MetadataSource} from '../external-views/externalMetadataSourceUtils';
+
+const InternalMetadataSourceContext = React.createContext({});
+
+export const InternalMetadataSourceProvider = ({children}) => {
+    const {metadataSources = [], error, loading, refresh} = useContext(MetadataSourceContext);
+
+    const internalMetadataSource: MetadataSource = metadataSources.find(source => source.path === null);
```

Comment:
could you please explain why we accept the first one with path = null?

Reply:
I described it in the documentation - path is the url of"|Fairspace instance to connect to. If the url is not specified, the metadata source will be treated as the internal one.
| *Important!* There should only be a single configuration of internal metadata (only the first one will not be ignored)."